verifier.go

```diff
@@ -0,0 +1,110 @@
+package authcontrol
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/jwtauth/v5"
+	"github.com/lestrrat-go/jwx/v2/jwa"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+)
+
+const DefaultAlgorythm = string(jwa.HS256)
+
+// AuthProvider is an interface for getting JWTAuth from a request
+type AuthProvider interface {
+	GetAuth(req *http.Request, options ...jwt.ValidateOption) (*jwtauth.JWTAuth, error)
```

Comment:
Req is not used at all. Why not to define interface as this? Or am i missing something?

```suggestion
	GetAuth(options ...jwt.ValidateOption) (*jwtauth.JWTAuth, error)
```

Reply:
`ProjectProvider` implementation uses it